catalogue/bpf/tests/IRIW+fencembonceonces+OnceOnce.litmus

```diff
@@ -0,0 +1,35 @@
+BPF IRIW+fencembonceonces+OnceOnce
+
+(*
+ * Result: Never
+ *
+ * Test of independent reads from independent writes with a fence between
+ * each pairs of reads.  In other words, is a full barrier sufficient to
+ * cause two different reading processes to agree on the order of a pair of
+ * writes, where each write is to a different variable by a different
+ * process?  This litmus test exercises LKMM's "propagation" rule.
+ *
+ * Atomic ops with a fetch act as full barriers.
+ *)
+
+{
+0:r0=x;
+
+1:r0=x;
+1:r1=y;
+1:r5=tmp;
+
+2:r1=y;
+
+3:r0=x;
+3:r1=y;
+3:r5=tmp2;
+}
+
+P0		     | P1		                         | P2                   | P3		                              ;
+*(u32 *)(r0 + 0) = 1 | r2 = *(u32 *)(r0 + 0)                     | *(u32 *)(r1 + 0) = 1 | r2 = *(u32 *)(r1 + 0)                       ;
+                     | r6 = atomic_fetch_add((u64*)(r5 + 0), r6) |                      | r6 = atomic_fetch_add((u64*)(r5 + 0), r6)   ;
```

Comment:
You are adding `r6`, but the value of this register is not initialized. Other tests have the same problem.

Reply:
Yes, `atomic_fetch_add` is just being used here to enforce ordering because BPF doesn't have a real fence instruction. r6 will be `0` by default and that is okay for this usage.

Reply:
Fixed this and similar usages of r6 but initialising it to 0